### ISSUE TITLE
config parsing fails with "Empty subsection name"

### ISSUE THREAD

**Issue by srijan-paul** [go-git/go-git#1197]
If the `.git/config` file has an empty subsection name like this:

```toml
[remote ""]
url = "foo"
fetch = "bar"
```

The official git implementation works fine, but go-git fails to parse the config with an error that says "empty subsection name".
Found out while using go-git at work to analyze git repos.

I'd be happy to send a patch to make go-git behave the same as git.

Minimal reproduction:

```go
package main

import "github.com/go-git/go-git/v5"

func must(err error) {
	if err != nil {
		panic(err)
	}
}

func main() {
       // somerepo/.git/config should have an empty toml subsection.
	repo, err := git.PlainOpen("./somerepo")
	must(err)

	remotes, err := repo.Remotes()
	must(err) // panics here

	for _, remote := range remotes {
		println(remote.Config().Name)
	}
```

**Comment by CLBRITTON2** [comment]
I can probably take a look at this

**Comment by CLBRITTON2** [comment]
@pjbgf I don't think it would be incredibly difficult to adapt `go-git/gcfg` to support. 
I didn't dive too deep but it seemed like removing the empty name check/error and it's test case from gcfg allowed tests to pass.
I'd have to look into what go-git tests would need modifying to allow empty subsection names.
That would at least allow empty subsections to exist in a config without throwing errors.
Adapting to match upstream `git` might be a little bit more involved. I believe both config parsing and encoding would have to be modified to capture options that fall under an empty subsection name.
I'll continue looking into this.
